### ISSUE TITLE
smbios 3.2 Formatted section length of structure MemoryDevice with handle 59 should be 84 bytes

### ISSUE THREAD

**Issue by ldfsilva** [jcreekmore/dmidecode#17]
The `MemoryDevice` length enforcement in [017_memory_device](https://github.com/jcreekmore/dmidecode/blame/master/src/structures/017_memory_device.rs#L403-L405) leads to `InvalidFormattedSectionLength` errors on systems running smbios `3.2`.

The aforementioned enforcement requires a length of `0x54` (84 bytes), whereas the system reports a length of `0x28` (40 bytes), as shown in [1].

I dropped the enforcement to test how it would behave and it resulted in the parsing of the data as expected. I'm curious to why we have it there, @jcreekmore do you recollect the motivation behind it?

I also looked at [smbios 3.2 spec](https://www.dmtf.org/sites/default/files/standards/documents/DSP0134_3.2.0.pdf) (ie: page 94) that states that: 

> "Length of the structure, 15h for version 2.1, 1Bh for version 2.3, 1Ch for version 2.6, 22h for version 2.7, 28h for version 2.8, **at least 34h for version 3.2 and later**"

I'm not sure whether this is a must or not, considered the system in question has 0x28 and `dmidecode` (linux binary) parses it correctly. Checking that I haven't found an enforcement on specific size, instead it seems to be done in a range, eg:

Excerpt from [dmidecode.c](https://github.com/mirror/dmidecode/blob/master/dmidecode.c#L4550).
```
			if (h->length < 0x28) break;
```

[1] Excerpt from `dmidecode` in `smbios 3.2`
```
# dmidecode 3.2
Getting SMBIOS data from sysfs.
SMBIOS 3.2.0 present.

Handle 0x003B, DMI type 17, 40 bytes
	Header and Data:
		11 28 3B 00 39 00 FE FF 48 00 40 00 00 20 09 00
		01 02 1A 80 20 6A 0A 03 04 05 06 01 00 00 00 00
		60 09 B0 04 B0 04 B0 04
```

**Comment by jcreekmore** [comment]
Boy, that is a mess. So, based on the spec, I would say that the check should be looking for < 0x34 and not exactly 0x54. I assume that the 0x54 was a set of typos. However, as you have found (and obviously the C file before you), an appropriate lower bound should be 0x28. Most of those size checks don’t have to be there since everything from configured memory speed on is Optional. 

If you want to make a PR changing that to a lower bound of 0x28, that would be great. I don’t have a system with smbios 3.2 at hand to test with.

**Comment by ldfsilva** [comment]
Sounds good, I'll put up a PR. Thanks for the quick response.

**Comment by jcreekmore** [comment]
Closed by #18
